src/plugins/vis_type_timeseries/server/lib/get_fields.ts

```diff
@@ -44,21 +44,24 @@ export async function getFields(
   requestContext: RequestHandlerContext,
   request: OpenSearchDashboardsRequest,
   framework: Framework,
-  indexPattern: string
+  indexPattern: string,
+  dataSourceId: string | null
```

Comment:
can dataSourceId be `dataSourceId?: string` or is it can be set as null somewhere?

Reply:
The `dataSourceId` is set as `string | null` in the router schema, which will call `getFields()`

Reply:
+1 to @BionIT. Instead of `null` can we make optional parameter? If this is just because of router schema, then I would prefer fixing in router schema instead of workaround to support that. How much is the effort?

Reply:
Fixed, after some investigation, it's possible to set a schema as optional via `schema.maybe()`